src/Symfony/Component/Lock/Key.php

```diff
@@ -70,4 +71,27 @@ public function getState($stateKey)
     {
         return $this->state[$stateKey];
     }
+
+    public function resetExpiringDate()
+    {
+        $this->expiringDate = null;
+    }
+
+    /**
+     * @param \DateTimeImmutable $expiringDate
+     */
+    public function reduceExpiringDate(\DateTimeImmutable $expiringDate)
+    {
+        if (null === $this->expiringDate || $this->expiringDate > $expiringDate) {
```

Comment:
When several store are combined, they can define different expiration date, the goal is to take the shorter one. cf `if (.. &&  $this->expiringDate > $expiringDate) { set value}`

Indeed, reducing a date does not really make sens. What's about `reduceLifeLength` or `advanceExpiringDate`?